shared/modules/selectors/smart-transactions.ts

```diff
@@ -89,13 +94,21 @@ const getIsAllowedRpcUrlForSmartTransactions = (
   return rpcUrl?.hostname?.endsWith('.infura.io');
 };
 
+const hasNonZeroBalance = (state: SmartTransactionsMetaMaskState) => {
+  const selectedAccount = getSelectedAccount(
+    state as unknown as MultichainState,
```

Comment:
Oof. These types are a mess. This `MultichainState` type doesn't belong here (there is nothing multichain-specific about this state or what we're doing with it), but I see that the selector expects it. I'll follow up separately about this.